src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -471,6 +471,32 @@ public static IEnumerable<object[]> GetAsync_IPBasedUri_Success_MemberData()
             }
         }
 
+        [Theory]
+        [InlineData("http://", "[::1234]")]
+        [InlineData("http://", "[::1234]:8080")]
+        public async Task GetAsync_IPv6AddressInHostHeader_CorrectlyFormatted(string scheme, string host)
```

Comment:
I assume this test fails without your product change?  Just checking.

Reply:
Yes.

Reply:
Without product change:

```
        Assert.Contains() Failure
        Not found: Host: [::1234]
        In value:  List<String> ["GET http://::1234/ HTTP/1.1", "Host: ::1234"]
```

Reply:
Only semi-related, is the Uri we're sending to the proxy correctly formatted, or does it also need [] around the IPv6 address?

Reply:
On WinHttpHandler, we will append [] around the IPv6 address.
"GET http://[::1234]/ HTTP/1.1", "Proxy-Connection: Keep-Alive", "Host: [::1234]"

On SocketsHttpHandler, currently we don't:
"GET http://::1234/ HTTP/1.1", "Host: [::1234]"

Per RFC (https://tools.ietf.org/html/rfc3986#section-3.2.2), I think we should do that for the URI we are sending as well.